libs/crypto/src/crypto.cpp

```diff
@@ -20,121 +20,22 @@
-#include "util/filesystem.hpp"
-
+#include "crypto/openssl/crypto.hpp"
+#endif
 namespace concord::crypto {
-using std::array;
-using std::pair;
-using std::string;
-using concord::Byte;
-using concord::crypto::KeyFormat;
-using concord::crypto::openssl::UniquePKEY;
-using concord::crypto::openssl::UniqueECKEY;
-using concord::crypto::openssl::UniquePKEYContext;
-using concord::crypto::openssl::UniqueBIO;
-using concord::crypto::openssl::OPENSSL_SUCCESS;
-using concord::crypto::Ed25519PrivateKeyByteSize;
-using concord::crypto::Ed25519PublicKeyByteSize;
-
-pair<string, string> generateECKeyPair(int id, const KeyFormat fmt) {
-  constexpr size_t maxKeySize = 2048;
-
-  UniqueBIO outbio;
-  openssl::UniqueECKEY myecc{EC_KEY_new_by_curve_name(id)};
-
-  ConcordAssertEQ(OPENSSL_SUCCESS, EC_KEY_generate_key(myecc.get()));
-  UniquePKEY pkey{EVP_PKEY_new()};
-  EVP_PKEY_assign_EC_KEY(pkey.get(), myecc.get());
-
-  array<Byte, maxKeySize> privKey;
-  array<Byte, maxKeySize> pubKey;
-  size_t privKeyLen{maxKeySize};
-  size_t pubKeyLen{maxKeySize};
-
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_private_key(pkey.get(), privKey.data(), &privKeyLen));
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_public_key(pkey.get(), pubKey.data(), &pubKeyLen));
-
-  pair<string, string> keyPair(boost::algorithm::hex(string(reinterpret_cast<char*>(privKey.data()), privKeyLen)),
-                               boost::algorithm::hex(string(reinterpret_cast<char*>(pubKey.data()), pubKeyLen)));
-
-  if (KeyFormat::PemFormat == fmt) {
-    keyPair = EdDSAHexToPem(keyPair);
-  }
-  return keyPair;
-}
-
-pair<string, string> generateEdDSAKeyPair(const KeyFormat fmt) {
-  UniquePKEY edPkey;
-  UniquePKEYContext edPkeyCtx(EVP_PKEY_CTX_new_id(NID_ED25519, nullptr));
-
-  ConcordAssertNE(edPkeyCtx, nullptr);
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_keygen_init(edPkeyCtx.get()));
-  EVP_PKEY* keygenRet = nullptr;
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_keygen(edPkeyCtx.get(), &keygenRet));
-  edPkey.reset(keygenRet);
-
-  array<Byte, Ed25519PrivateKeyByteSize> privKey;
-  array<Byte, Ed25519PublicKeyByteSize> pubKey;
-  size_t keyLen{Ed25519PrivateKeyByteSize};
-
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_private_key(edPkey.get(), privKey.data(), &keyLen));
-  ConcordAssertEQ(keyLen, Ed25519PrivateKeyByteSize);
-  keyLen = Ed25519PublicKeyByteSize;
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_public_key(edPkey.get(), pubKey.data(), &keyLen));
-  ConcordAssertEQ(keyLen, Ed25519PublicKeyByteSize);
 
-  pair<string, string> keyPair(
-      boost::algorithm::hex(string(reinterpret_cast<char*>(privKey.data()), Ed25519PrivateKeyByteSize)),
-      boost::algorithm::hex(string(reinterpret_cast<char*>(pubKey.data()), Ed25519PublicKeyByteSize)));
-
-  if (KeyFormat::PemFormat == fmt) {
-    keyPair = EdDSAHexToPem(keyPair);
-  }
-  return keyPair;
+std::pair<std::string, std::string> generateEdDSAKeyPair(const KeyFormat fmt) {
+#ifdef USE_OPENSSL
```

Comment:
Since there is no other implementation, add an else clause with an indicative error

Reply:
done